docs/examples/Custom_Pandas_Dataloader.py

```diff
@@ -0,0 +1,280 @@
+as needed, including .csv, JSON, XML, Parquet, HDF5, .txt, .zip and more. 
+The resulting DataFrame can then simply be fed into the defined 
+`DataFrameGroundTruthReader` or `DataFrameDetectionReader` for further processing
+in Stonesoup as required. 
+"""
+
+# %%
+# Software dependencies
+# ---------------------
+# Before beginning this example, you need to ensure that Pandas is installed, 
+# which is a fast, powerful and flexible open-source data analysis tool in Python.
+# The easiest way to install pandas (if not done so already), is to run pip install
+# from a terminal window within the desired environment:
+#
+# .. code::
+#
+#     pip install pandas
+
+# %%
+# The main dependencies and imports for this example are included below:
+
+import numpy as np
+import os
+import pandas as pd
+
+from stonesoup.base import Property
+from stonesoup.buffered_generator import BufferedGenerator
+from stonesoup.reader.base import GroundTruthReader, DetectionReader, Reader
+from stonesoup.types.groundtruth import GroundTruthPath, GroundTruthState
+
+from typing import Sequence, Collection
+
+from datetime import datetime, timedelta
+from dateutil.parser import parse
+
+# %%
+# Data Frame Reader
+# ^^^^^^^^^^^^^^^^^
+# Similarly to Stonesoups :class:`~._CSVFrameReader`, we'll define a `_DataFrameReader`
+# class that inherits from the base :class:`~.Reader` class to read a DataFrame containing
+# state vector fields, a time field, and additional metadata fields (all other columns
+# by default). The only difference between this class and the :class:`~._CSVFrameReader` 
+# class is that we have no path attribute (the DataFrame is already loaded in memory).
+
+class _DataFrameReader(Reader):
+    state_vector_fields: Sequence[str] = Property(
+        doc='List of columns names to be used in state vector')
+    time_field: str = Property(
+        doc='Name of column to be used as time field')
+    time_field_format: str = Property(
+        default=None, doc='Optional datetime format')
+    timestamp: bool = Property(
+        default=False, doc='Treat time field as a timestamp from epoch')
+    metadata_fields: Collection[str] = Property(
+        default=None, doc='List of columns to be saved as metadata, default all')
+
+    def _get_metadata(self, row):
+        if self.metadata_fields is None:
+            local_metadata = dict(row)
+            for key in list(local_metadata):
+                if key == self.time_field or key in self.state_vector_fields:
+                    del local_metadata[key]
+        else:
+            local_metadata = {field: row[field]
+                              for field in self.metadata_fields
+                              if field in row}
+        return local_metadata
+
+    def _get_time(self, row):
+        if self.time_field_format is not None:
+            time_field_value = datetime.strptime(row[self.time_field], self.time_field_format)
+        elif self.timestamp is True:
```

Comment:
```suggestion
        elif self.timestamp:
```
You don't need the "is True" for a bool.